generator/src/main/java/com/graphicsfuzz/generator/mutateapi/Value.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 The GraphicsFuzz Project Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.graphicsfuzz.generator.mutateapi;
+
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.type.BasicType;
+import com.graphicsfuzz.common.ast.type.Type;
+import com.graphicsfuzz.generator.fuzzer.Fuzzer;
+import com.graphicsfuzz.generator.semanticschanging.LiteralFuzzer;
+import java.util.List;
+import java.util.Optional;
+
+public interface Value {
+
+  Type getType();
+
```

Comment:
Thinking about this more, can we change "valueIsKnown" to be called "valueIsUnknown", so that it returns true if and only if the value is totally unknown?  (This will require flipping all uses of the method).

My reasoning is that in the case of a composite value, we currently say that "valueIsKnown" is true if there is some list of values available, even if some of the values in the list are themselves unknown or contain unknown values inside them.  So really by saying "valueIsKnown" we mean "we know something about this value", not "it is completely known".

I think changing things so that "valueIsUnknown" returns true if the value is *totally* unknown would be clearer.

Reply:
Fixed